client/src/components/SigninButton.svelte

```diff
@@ -1,19 +1,151 @@
 <script lang="ts">
   import Modal from "./Modal.svelte";
   import { authStore as user } from "$stores/auth";
+  import { loginWithGoogle, loginWithMicrosoft, loginWithEmail, onSignOut, createUser } from "$lib/firebase/auth";
 
   // check if the user is logged in with getAuth
   let openSignInModal = false;
+  let email = "";
+  let password = "";
+  let confirmPass = "";
+  let clickedButton = false;
+  let clickedButton2 = false;
+  let errorMessage = "";
+  async function createAccount() {
+    if (password == "") {
+      return;
+    }
+    if (password != confirmPass) {
+      errorMessage = "Password does not match";
+      password = "";
+      confirmPass = "";
+      return;
+    }
+    if (email == "") {
+      return;
+    } else {
+      try {
+        await createUser(email, password);
+        errorMessage = "";
+      } catch (err) {
+        errorMessage = "Invalid Username or Password";
+      }
+    }
+
+    if (errorMessage == "") {
+      openSignInModal = false;
+    }
+    email = "";
+    password = "";
+    confirmPass = "";
+  }
+  async function submitLogin() {
+    if (password == "") {
+      return;
+    }
+    if (email == "") {
+      return;
+    } else
+      try {
+        await loginWithEmail(email, password);
+        errorMessage = "";
+      } catch (err) {
+        errorMessage = "Invalid Username or Password";
+      }
+    if (errorMessage == "") {
+      openSignInModal = false;
+    }
+    password = "";
+    email = "";
+  }
+  function signInWithEmailAndPassword() {
+    clickedButton = !clickedButton;
+    clickedButton2 = false;
+  }
+  function createWithEmailAndPassword() {
+    clickedButton2 = !clickedButton2;
+    clickedButton = false;
+  }
+  async function googleLogin() {
+    await loginWithGoogle();
+    openSignInModal = false;
+  }
+
+  async function microsoftLogin() {
+    await loginWithMicrosoft();
+    openSignInModal = false;
+  }
 </script>
 
 <!-- Sign In Modal -->
-<Modal
-  open={openSignInModal}
-  onClosed={() => {
-    openSignInModal = false;
-  }}
->
+<Modal open={openSignInModal}>
   <!-- TODO: Sign In Modal Content Here -->
```

Comment:
we probobly don't need this TODO anymore

Reply:
resolved